### PR TITLE
Remove vendoring from project

### PR DESCRIPTION
**Pull request by afritzler**
**How to categorize this PR?**
/area dev-productivity
/kind enhancement
/platform openstack

**What this PR does / why we need it**:

***This is a WIP PR to illustrate how to remove vendoring from Gardener extension projects***

- Breaking webhook coding in Gardener has not been addressed here. 
- Fix `.ci` based vendor dependencies by copying the relevant CI files into a `hack` subfolder
- Fix references in `Makefile` by using `go list` to determine where to find the Gardener `hack` scripts
- Fix `go generate` markers to use the correct script location by injecting the `GARDENER_HACK_DIR` location via `Makefile`
- Use `go mod download` in `Dockerfile` to prefetch deps
- Use `etcd-druid-api` instead of `etcd-druid`

/cc @kon-angelo @MartinWeindel 

**Release note**:
```other developer
Remove vendoring from project
```

### CODE REVIEW

File: .ci/component_descriptor [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env bash
 
-"$(dirname $0)"/../vendor/github.com/gardener/gardener/hack/.ci/component_descriptor "$(dirname $0)"/..
+"$(dirname $0)"/hack/gardener_component_descriptor "$(dirname $0)"/..
```
Comment (MartinWeindel) [inline .ci/component_descriptor:3]: fix filename
```suggestion
"$(dirname $0)"/hack/component_descriptor "$(dirname $0)"/..
```

File: Makefile [hunk 2]
```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+GARDENER_HACK_DIR           := $(shell go list -m -f "{{.Dir}}" github.com/gardener/gardener)/hack
```
Comment (MartinWeindel) [inline Makefile:15]: ensure gardener module directtory available
```suggestion
ENSURE_GARDENER_MOD         := $(shell go get github.com/gardener/gardener@$$(go list -m -f "{{.Version}}" github.com/gardener/gardener))
GARDENER_HACK_DIR           := $(shell go list -m -f "{{.Dir}}" github.com/gardener/gardener)/hack
```

**Review by MartinWeindel — CHANGES_REQUESTED** [review]
suggestions to fix build errorrs

File: .ci/hack/component_descriptor [hunk 3]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
```
Comment (vpnachev) [inline .ci/hack/component_descriptor]: Are those hack scripts now copied and they have to be maintained in each repository using them or they are somehow synced from a single source repository?

Comment (afritzler) [reply .ci/hack/component_descriptor]: They are copied via `make tidy`. Essentially that happened in the past via `make revendor` and stored in the `vendor` folder. Unfortunately in the Councourse CI environment we can't rely on a golang environment to be present so we can't reference them dynamically via `go mod`/`go list`.

Comment (vpnachev) [reply .ci/hack/component_descriptor]: That is very nice, thanks for the clarification!

**Comment by afritzler** [comment]
/ref https://github.com/gardener/etcd-druid/issues/734

Currently `etcd-druid` is on a super old Gardener version (1.76). This leads to some weird behaviour when updating the latest Gardener commit version. Adding a ref here to keep track of this issue.

**Comment by timuthy** [comment]
As discussed, replaced by #704
/close

**Pull request closed without merging**